typescript/test/deposit-sweep.test.ts

```diff
@@ -406,38 +403,30 @@ describe("Sweep", () => {
           expect(txJSON.inputs.length).to.be.equal(2)
 
           const p2shInput = txJSON.inputs[0]
-          expect(p2shInput.prevout.hash).to.be.equal(
+          expect(p2shInput.hash).to.be.equal(
             depositSweepWithNoMainUtxoAndWitnessOutput.deposits[0].utxo.transactionHash.toString()
           )
-          expect(p2shInput.prevout.index).to.be.equal(
+          expect(p2shInput.index).to.be.equal(
             depositSweepWithNoMainUtxoAndWitnessOutput.deposits[0].utxo
               .outputIndex
           )
           // Transaction should be signed. As it's not SegWit input, the `witness`
           // field should be empty, while the `script` field should be filled.
-          expect(p2shInput.witness).to.be.equal("00")
+          expect(p2shInput.witness).to.be.empty
           expect(p2shInput.script.length).to.be.greaterThan(0)
-          // Input's address should be set to the address generated from deposit
-          // script hash
-          expect(p2shInput.address).to.be.equal(testnetDepositScripthashAddress)
 
           const p2wshInput = txJSON.inputs[1]
-          expect(p2wshInput.prevout.hash).to.be.equal(
+          expect(p2wshInput.hash).to.be.equal(
             depositSweepWithNoMainUtxoAndWitnessOutput.deposits[1].utxo.transactionHash.toString()
           )
-          expect(p2wshInput.prevout.index).to.be.equal(
+          expect(p2wshInput.index).to.be.equal(
             depositSweepWithNoMainUtxoAndWitnessOutput.deposits[1].utxo
               .outputIndex
           )
           // Transaction should be signed. As it's a SegWit input, the `witness`
           // field should be filled, while the `script` field should be empty.
           expect(p2wshInput.witness.length).to.be.greaterThan(0)
           expect(p2wshInput.script.length).to.be.equal(0)
-          // Input's address should be set to the address generated from deposit
-          // witness script hash
-          expect(p2wshInput.address).to.be.equal(
-            testnetDepositWitnessScripthashAddress
-          )
```

Comment:
Sadly, I had to remove the input's address check.
The reason is that it's rather difficult and even in the original `bcoin` implementation it was described as "guessing". If we really want to preserve this check we would have to port some code from `bcoin`.